docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,578 @@
+
+* Create, list, watch verbs for all resources that are a part of the install (cannot be scoped to specific resource names).
+  - Permissions to create and manage CustomResourceDefinitions
+  - Permissions to create any other manifest objects
+  - Rules to manage any other cluster-scoped resources
+  - All the rules defined in the CSV under `.spec.install.clusterPermissions` and `.spec.install.permissions`
+  - Permissions for namespace-scoped resources.
+  - These are specified with ClusterRole + ClusterRoleBinding, ClusterRole + RoleBinding, or Role + RoleBinding.
+
+* Get, update, patch, delete verbs for all resources that are a part of the install (can be scoped to specific resource names)
+  - Permissions for cluster-scoped resources
+  - Permissions for namespace-scoped resources
+  - These are created using ClusterRole + ClusterRoleBinding, ClusterRole + RoleBinding, or Role + RoleBinding
+
+* Permissions to create any ClusterRole, ClusterRoleBinding, Role, RoleBinding resources required by the extension.
+  - All the same rules defined in the ClusterRole and Role resources
+  - Escalate and bind verbs for ClusterRole, ClusterRoleBinding, Role, RoleBinding resources
+  - Permissions to create the necessary roles and rolebindings for the controller to be able to perform its job
+
+* Update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+
+* Permissions to create the controller deployment, this corresponds to the rules to manage the
+  deployment defined in the ClusterServiceVersion
+
+### Derive minimal RBAC
+
+As an example, consider a cluster extension that needs to query OpenShift users and groups as part of its controller logic and specifies the below cluster permissions in its ClusterServiceVersion:
+
+```yml
+clusterPermissions:
+        - rules:
+            - apiGroups:
+                - user.openshift.io
+              resources:
+                - users
+                - groups
+                - identities
+              verbs:
+                - get
+                - list
+```
+
+In addition to cluster permissions, it specifies these additional permissions to manage itself:
+
+```yml
+permissions:
+        - rules:
+            - apiGroups:
+                - apps
+              resourceNames:
+                - <cluster-extension-name>
+              resources:
+                - deployments/finalizers
+              verbs:
+                - update
+            - apiGroups:
+                - apps
+              resources:
+                - deployments
+              verbs:
+                - create
+                - delete
+                - get
+                - list
+                - patch
+                - update
+                - watch
+```
+
+Below is the translation of the above cluster permissions into ClusterRole and ClusterRoleBinding for your cluster extension:
```

Comment:
Suggested text, `Here are the example of ClusterRolend and ClusterRoleBinding  for the above cluster permissions`.